% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/metaAnalysisESpath.R
\name{metaAnalysisESpath}
\alias{metaAnalysisESpath}
\title{Performing Gene Set Enrichment Meta-analysis}
\usage{
metaAnalysisESpath(
  objectMApath = NULL,
  effectS = NULL,
  measure = c("limma", "SMD", "MD"),
  WithinVarCorrect = TRUE,
  typeMethod = c("FEM", "REM"),
  missAllow = 0.3,
  proportionData = 1
)
}
\arguments{
\item{objectMApath}{A list of list. Each list contains two elements.
The first element is the Gene Set matrix (gene sets in rows and samples in
columns) and the second element is a vector of zeros and ones that represents
the state of the different samples of the Gene Sets matrix.
0 represents one group (controls) and 1 represents the other group (cases).}

\item{effectS}{A list of two elements. The first element is a dataframe
with gene sets in rows and studies in columns.
Each component of the dataframe is the effect of a gene set in a study.
The second element of the list is also a dataframe
with the same structure, but in this case each component of the dataframe
represent the variance of the effect of a gene set in a study. This argument
should be only used in the case that objectMApath argument is null.}

\item{measure}{A character string that indicates the type of effect size to
be calculated. The options are "limma", "SMD" and "MD". The default value
is "limma". See details for more information.}

\item{WithinVarCorrect}{A logical value that indicates if the within variance
correction should be applied. The default value is TRUE. See details for more
information.}

\item{typeMethod}{A character that indicates the method to be performed.
See "Details"for more information}

\item{missAllow}{a number that indicates the maximum proportion of missing
values allowed in a sample. If the sample has more proportion of missing
values the sample will be eliminated. In the other case the missing values
will be imputed using the K-NN algorithm.}

\item{proportionData}{The minimum proportion of datasets in which a gene
must be contained to be included. By default, the gene must be contained
in at least half of the datasets}
}
\value{
A dataframe with the meta-analysis results. For more information
see the package vignette.
}
\description{
It performs Gene Sets Enrichment meta-analysis by applying Effects size
combination methods
}
\details{
The different meta-analysis methods that can be applied are:
\enumerate{
 \item "FEM": Fixed Effects model
 \item "REM": Random Effects model
    }
}
\examples{

results <- metaAnalysisESpath(objectMApath = objectMApath_sim,
    measure = "limma, typeMethod = "REM)

}
\references{
Daniel Toro-Domínguez, Juan Antonio Villatoro-García,
Jordi Martorell-Marugán, Yolanda Román-Montoya, Marta E Alarcón-Riquelme,
Pedro Carmona-Sáez (2020).
A survey of gene expression meta-analysis: methods and applications,
Briefings in Bioinformatics, bbaa019,
\url{https://doi.org/10.1093/bib/bbaa019}

Borenstein, M. (2009). Effect sizes for continuous data. In H. Cooper,
L. V. Hedges, & J. C. Valentine (Eds.),
The handbook of research synthesis and meta-analysis (2nd ed., pp. 221–235).
New York: Russell Sage Foundation.

Hedges, L. V. (1981). Distribution theory for Glass's estimator of effect
size and related estimators. Journal of Educational Statistics, 6(2),
107–128. \url{⁠https://doi.org/10.3102/10769986006002107}

Lin L, Aloe AM (2021). Evaluation of various estimators for standardized mean
difference in meta-analysis. Stat Med. 2021 Jan 30;40(2):403-426.
\url{https://doi.org/10.1002/sim.87811}
}
\seealso{
\code{\link{calculateESpath}}
}
\author{
Juan Antonio Villatoro Garcia,
\email{juanantoniovillatorogarcia@gmail.com}
}
